components/director/internal/domain/formation/service.go

```diff
@@ -1186,24 +1197,25 @@ func (s *service) resynchronizeFormationAssignmentNotifications(ctx context.Cont
 	}
 
 	alreadyProcessedFAs := make(map[string]bool, 0)
+	assignmentMappingPairs := append(assignmentMappingNoNotificationPairs, append(assignmentMappingSyncPairs, assignmentMappingAsyncPairs...)...)
```

Comment:
We should add a comment with description why we separate the pairs and then append them. Also it should be added in the FA service as well. 

In the new format e2e tests for the config mutator there are two comments that describes two cases based on the notification execution order - the comment should be deleted